### PR TITLE
Catch all JS exceptions at the beginning

### PR DESCRIPTION
**Pull request by pksunkara**
Fixes the crash in #45

### CODE REVIEW

File: src/parser.js [hunk 1]
```diff
@@ -30,6 +30,10 @@ const ANNOTATIONS = {
     code: 4,
     fragment: 'swagger-validation',
   },
+  UNCAUGHT_ERROR: {
```
Comment (kylef) [inline src/parser.js]: Remember to update the table in the READMe including the new code https://github.com/apiaryio/fury-adapter-swagger#parser-codes

File: src/parser.js [hunk 2]
```diff
@@ -148,29 +152,36 @@ export default class Parser {
...
+        // Swagger has a paths object to loop through that describes resources
+        _.each(_.omit(swagger.paths, isExtension), (pathValue, href) => {
+          this.handleSwaggerPath(pathValue, href);
+        });
 
-      done(null, this.result);
+        return done(null, this.result);
+      } catch (exception) {
+        this.createAnnotation(ANNOTATIONS.UNCAUGHT_ERROR, null,
+          ('Problem converting the input into refract due to swagger support still being in beta'));
```
Comment (kylef) [inline src/parser.js]: I don't think this message is very user friendly.

What about:

> There was a problem converting the Swagger document, the document may be using unsupported features in the Swagger Beta.

**Comment by pksunkara** [comment]
Updated.

File: src/parser.js [hunk 3]
```diff
@@ -148,29 +152,36 @@ export default class Parser {
...
+          this.handleSwaggerPath(pathValue, href);
+        });
 
-      done(null, this.result);
+        return done(null, this.result);
+      } catch (exception) {
+        this.createAnnotation(ANNOTATIONS.UNCAUGHT_ERROR, null,
+          ('There was a problem converting the Swagger document, the document may be using unsupported features in the Swagger Beta'));
+
+        return done(new Error(exception.message), this.result);
```
Comment (kylef) [inline src/parser.js]: Should we return the result as it is? Perhaps it will have broken / unknown state due to the exception?

Comment (pksunkara) [inline src/parser.js]: It will be in a good state. It will have stuff until the stuff got added and that's all. You can check the test case.

**Comment by Baggz** [comment]
@pksunkara @kylef 

I'll leave the final review for @danielgtaylor 

But as far as I see this is safe—it won't break Editor Kit (used to display semantic issues and beta warnings in the API Editor), nor Apiary.

**Comment by danielgtaylor** [comment]
LGTM guys :+1:

**Pull request merged**